tools/cpp/windows_cc_configure.bzl

```diff
@@ -93,6 +93,36 @@ def find_vc_path(repository_ctx):
     _auto_configure_warning_maybe(repository_ctx, "'BAZEL_VC' is not set, " +
                                                   "start looking for the latest Visual C++ installed.")
 
+    if "BAZEL_VC_TOOL" in repository_ctx.os.environ:
+        _auto_configure_warning_maybe(repository_ctx, "'BAZEL_VC_TOOL' is set, " +
+                               "but 'BAZEL_VC' and 'BAZEL_VS' are not. " + 
+                               "This is not recomended and can " +
+                               "lead to incorrect VC++ detection and configuration.")
+        vc_tool_path = repository_ctx.os.environ["BAZEL_VC_TOOL"]
+        # As neither 'BAZEL_VC' nor 'BAZEL_VS' are set, we assume BAZEL_VC_TOOL looks like that: %VS_PATH%\VC\<path_to_tool>
+        # path_to_tool is different for different VS versions, but we don't care
+        # We DO NOT search in any other places as we can easily 
+        # get inconsistancy with VC++ tools and VCVARSALL.BAT (assuming multiple versions of VS are installed)
+        vc_tool_path = vc_tool_path.replace('\\', '/').strip()
+        vc_pos = vc_tool_path.rfind('/VC/')
+        if vc_tool_path == 'VC' or vc_tool_path.endswith('/VC'):
+            vc_tool_path = vc_tool_path + '/'
+        elif vc_pos != -1:
+            vc_tool_path = vc_tool_path[:vc_pos + len('/VC/')]
+        elif vc_tool_path.startswith('VC/'):
+            vc_tool_path = 'VC/'
+        else:
+            vc_tool_path = None
+
+        if vc_tool_path:
+            _auto_configure_warning_maybe(repository_ctx, ("VC++ found at '%s'" % vc_tool_path))
+        else:
+             _auto_configure_warning_maybe(repository_ctx, "Bazel failed to find VC++. "+
+                "You need either specify any of 'BAZEL_VC' / 'BAZEL_VS' "  + 
+                "or make sure that 'BAZEL_VC_TOOL' contains subpath to 'VC' directory.")
+
+        return vc_tool_path
```

Comment:
This may be a relative path, see https://github.com/bazelbuild/bazel/pull/7686/files?file-filters%5B%5D=.bzl#diff-457360df3a98d4f91f49ccc5c8f3b35aR113

Is that OK?